crates/runtime/src/embeddings/execution_plan.rs

```diff
@@ -400,6 +407,75 @@ pub(super) async fn get_vectors(
     Ok(builder.finish())
 }
 
+/// Embed a [`StringArray`] using the provided [`Embed`] model with parallel processing.
+/// Similar to [`get_vectors`] but runs synchronously and processes embeddings in parallel
+/// across multiple threads using [`rayon::par_iter`]. The output is a [`FixedSizeListArray`],
+/// where each [`String`] gets embedded into a single [`f32`] vector.
+#[allow(clippy::cast_sign_loss, clippy::cast_possible_truncation)]
+pub(super) fn get_vectors_in_process<'a>(
+    arr: impl Iterator<Item = Option<&'a str>>,
+    model: &dyn Embed,
+    vector_length: i32,
+) -> Result<FixedSizeListArray, Box<dyn std::error::Error + Send + Sync>> {
+    let inputs: Vec<Option<String>> = arr.map(|o| o.map(ToString::to_string)).collect();
+
+    let mut builder = FixedSizeListBuilder::with_capacity(
+        PrimitiveBuilder::<Float32Type>::with_capacity(inputs.len() * (vector_length as usize)),
+        vector_length,
+        inputs.len(),
+    )
+    .with_field(Arc::new(Field::new("item", DataType::Float32, true)));
```

Comment:
```suggestion
    .with_field(Arc::new(Field::new("item", DataType::Float32, false)));
```

The inner floats shouldn't be nullable I believe

Reply:
I agree with you, but it seems the DF schema at this point has it set as `nullable: true`. To expedite the release, I've captured this in an issue and will revert this change:
https://github.com/spiceai/spiceai/issues/6911